cli/command/image/build.go

```diff
@@ -227,13 +239,56 @@ func runBuild(dockerCli *command.DockerCli, options buildOptions) error {
 		buildCtx, err = archive.TarWithOptions(contextDir, &archive.TarOptions{
 			Compression:     compression,
 			ExcludePatterns: excludes,
-			IncludeFiles:    includes,
```

Comment:
I just wanted it to explain why `ExcludePatterns` is being used instead of `IncludeFiles` for additional files that are added to the tar. According to @tonistiigi above, it's because `IncludeFiles` messes up the order of the files. That said, I don't know if we care about file ordering anymore.